# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/threads/SchedulerGroup.h
# Commit: 36b0646e8649
# Full Hash: 36b0646e86498662e770756ad54b1fd648a82184
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2020-04-07 21:44:02
# Regressor Bug: 1620594
# File Overlap Count: 1
# Description:
#   Bug 1620594 - Part 7: Remove TabGroup and SystemGroup. r=nika,bas
#   
#   TabGroup never really made any difference in which thread something go
#   dispatched to. This was the intended use, but development of TabGroups
#   with abstract main threads never made it that far. The good thing is
# ==============================================================================

diff -r a5ad3b179c92 -r 36b0646e8649 xpcom/threads/SchedulerGroup.h
--- a/xpcom/threads/SchedulerGroup.h	Tue Apr 07 15:17:14 2020 +0000
+++ b/xpcom/threads/SchedulerGroup.h	Tue Apr 07 15:17:47 2020 +0000
@@ -13,6 +13,7 @@
 #include "mozilla/Queue.h"
 #include "mozilla/TaskCategory.h"
 #include "mozilla/ThreadLocal.h"
+#include "mozilla/ThrottledEventQueue.h"
 #include "mozilla/TimeStamp.h"
 #include "nsCOMPtr.h"
 #include "nsISupportsImpl.h"
@@ -26,7 +27,6 @@
 class AbstractThread;
 namespace dom {
 class DocGroup;
-class TabGroup;
 }  // namespace dom
 
 #define NS_SCHEDULERGROUPRUNNABLE_IID                \
@@ -44,9 +44,9 @@
 //
 // A SchedulerGroup is an abstract class to represent a "group". Essentially the
 // only functionality offered by a SchedulerGroup is the ability to dispatch
-// runnables to the group. TabGroup, DocGroup, and SystemGroup are the concrete
+// runnables to the group. DocGroup, and SystemGroup are the concrete
 // implementations of SchedulerGroup.
-class SchedulerGroup : public LinkedListElement<SchedulerGroup> {
+class SchedulerGroup {
  public:
   SchedulerGroup();
 
@@ -79,20 +79,9 @@
   };
   friend class Runnable;
 
-  bool* GetValidAccessPtr() { return &mIsRunning; }
-
   static nsresult Dispatch(TaskCategory aCategory,
                            already_AddRefed<nsIRunnable>&& aRunnable);
 
-  virtual nsISerialEventTarget* EventTargetFor(TaskCategory aCategory) const;
-
-  // Must always be called on the main thread. The returned AbstractThread can
-  // always be used off the main thread.
-  AbstractThread* AbstractMainThreadFor(TaskCategory aCategory);
-
-  // This method performs a safe cast. It returns null if |this| is not of the
-  // requested type.
-  virtual dom::TabGroup* AsTabGroup() { return nullptr; }
 
   static nsresult UnlabeledDispatch(TaskCategory aCategory,
                                     already_AddRefed<nsIRunnable>&& aRunnable);
@@ -101,50 +90,19 @@
 
   static void MarkVsyncRan();
 
-  void SetIsRunning(bool aIsRunning) { mIsRunning = aIsRunning; }
-  bool IsRunning() const { return mIsRunning; }
-
-  struct EpochQueueEntry {
-    nsCOMPtr<nsIRunnable> mRunnable;
-    uintptr_t mEpochNumber;
-
-    EpochQueueEntry(already_AddRefed<nsIRunnable> aRunnable, uintptr_t aEpoch)
-        : mRunnable(aRunnable), mEpochNumber(aEpoch) {}
-  };
-
-  using RunnableEpochQueue = Queue<EpochQueueEntry, 32>;
-
-  RunnableEpochQueue& GetQueue(mozilla::EventQueuePriority aPriority) {
-    return mEventQueues[size_t(aPriority)];
-  }
-
- protected:
   static nsresult DispatchWithDocGroup(
       TaskCategory aCategory, already_AddRefed<nsIRunnable>&& aRunnable,
       dom::DocGroup* aDocGroup);
 
+ protected:
   static nsresult InternalUnlabeledDispatch(
       TaskCategory aCategory, already_AddRefed<Runnable>&& aRunnable);
 
-  // Implementations are guaranteed that this method is called on the main
-  // thread.
-  virtual AbstractThread* AbstractMainThreadForImpl(TaskCategory aCategory);
-
-  // Helper method to create an event target specific to a particular
-  // TaskCategory.
-  virtual already_AddRefed<nsISerialEventTarget> CreateEventTargetFor(
-      TaskCategory aCategory);
-
-  // Given an event target returned by |dispatcher->CreateEventTargetFor|, this
-  // function returns |dispatcher|.
-  static SchedulerGroup* FromEventTarget(nsIEventTarget* aEventTarget);
 
   static nsresult LabeledDispatch(TaskCategory aCategory,
                                   already_AddRefed<nsIRunnable>&& aRunnable,
                                   dom::DocGroup* aDocGroup);
 
-  void CreateEventTargets(bool aNeedValidation);
-
   // Shuts down this dispatcher. If aXPCOMShutdown is true, invalidates this
   // dispatcher.
   void Shutdown(bool aXPCOMShutdown);
@@ -154,10 +112,6 @@
   // Number of events that are currently enqueued for this SchedulerGroup
   // (across all queues).
   size_t mEventCount = 0;
-
-  nsCOMPtr<nsISerialEventTarget> mEventTargets[size_t(TaskCategory::Count)];
-  RefPtr<AbstractThread> mAbstractThreads[size_t(TaskCategory::Count)];
-  RunnableEpochQueue mEventQueues[size_t(mozilla::EventQueuePriority::Count)];
 };
 
 NS_DEFINE_STATIC_IID_ACCESSOR(SchedulerGroup::Runnable,